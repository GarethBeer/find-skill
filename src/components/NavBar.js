import React from "react";
import { Link } from "react-router-dom";
import "../styles/NavBar.css";
import MyLogo from "../MyLogo.png";

const NavBar = () => {
  return (
    <div className="navbar">
      <img className="navbar-logo" src={MyLogo} alt="logo" />
      <h2>ShareASkill</h2>
      <ul className="navbar-links">
        <li className="navbar-links-item">
          <Link to="/">View Skills</Link>
        </li>
        <li className="navbar-links-item">
          <Link to="/add-skills">Manage a Skill</Link>
        </li>
      </ul>
    </div>
  );
};

export default NavBar;
