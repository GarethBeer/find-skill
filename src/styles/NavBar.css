.navbar {
    display:flex;
    justify-content: flex-start;
    align-items: center;
    flex-wrap: wrap;
    background-color: white;
    border-bottom: 1px solid grey;
}
.navbar ul:last-child {
    margin-left: auto;
}
h2 {
    font-size: 30px;
    text-align: center;
    font-weight: 900;
    margin-top: 5vh;
    margin-bottom: 0em;
}
.navbar-logo {
    width: 7%;  
    height: 7%;  
    margin-right: 2%;
}
.navbar-links {
    display: flex;
    justify-content: flex-end;
    flex-wrap: nowrap;
    float: right;
    margin-block-start: 0em;
    padding-inline-start: 0px;
}
.navbar-links-item {
    font-weight: 700;
    font-size: 25px;
    margin: 0vh 3vh 0vh 3vh;
    display: inline-block;
} 
.navbar-links-item :hover {
    color: rgb(18, 122, 18)
}
.navbar-links-item > * {
    text-decoration: none;
    color: black;
}
@media only screen and (max-width: 560px) {
    .navbar {
        justify-content: center;
    }
    .navbar-logo{
        width: 20%;  
        height: 20%;
    }
    .navbar-links-item {
        font-size: 20px;
    }
    .navbar ul:last-child {
        margin-left: 0px;
    }
}