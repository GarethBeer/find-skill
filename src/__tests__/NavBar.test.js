import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  const setUp = () => {
    const { asFragment, getByRole } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );

    return {
      asFragment: asFragment(),
      linkToSkills: getByRole("link", {
        name: /view skill/i,
      }),
      linkToAddSkill: getByRole("link", {
        name: /manage a skill/i,
      }),
    };
  };

  let renderedComponent;

  beforeEach(() => {
    renderedComponent = setUp();
  });

  it("renders correctly to match the NavBar snapshot", () => {
    expect(renderedComponent.asFragment).toMatchSnapshot();
  });

  it("renders skills link properly", () => {
    expect(renderedComponent.linkToSkills).toHaveAttribute("href", "/");
  });

  it("renders add skill link properly", () => {
    expect(renderedComponent.linkToAddSkill).toHaveAttribute(
      "href",
      "/add-skills"
    );
  });
});
