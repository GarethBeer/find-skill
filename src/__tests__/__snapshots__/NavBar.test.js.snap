// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly to match the NavBar snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="logo"
      class="navbar-logo"
      src="MyLogo.png"
    />
    <h2>
      ShareASkill
    </h2>
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links-item"
      >
        <a
          href="/"
        >
          View Skills
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          href="/add-skills"
        >
          Manage a Skill
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
